dialogue-target/src/main/java/com/palantir/dialogue/Request.java

```diff
@@ -16,37 +16,203 @@
 
 package com.palantir.dialogue;
 
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.ImmutableSortedMap;
 import com.google.common.collect.Multimap;
+import com.palantir.logsafe.Preconditions;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
-import org.immutables.value.Value;
 
 /** Defines the parameters of a single {@link Call} to an {@link Endpoint}. */
-@DialogueImmutablesStyle
-@Value.Immutable
-public interface Request {
-    /** The HTTP headers for this request, encoded as a map of {@code header-name: header-value}. */
-    Map<String, String> headerParams();
+public final class Request {
```

Comment:
:+1: to hand-rolling - it seems much easier to control exactly which methods exist (and maintain back compat) when we write them ourselves vs have immutables magically generate them!